compiler/noirc_evaluator/src/ssa/ir/types.rs

```diff
@@ -79,6 +79,20 @@ impl Type {
         }
     }
 
+    pub(crate) fn contains_slice_element(&self) -> bool {
+        match self {
+            Type::Array(elements, _) => {
+                elements.iter().any(|element| element.contains_slice_element())
+            }
+            Type::Slice(_) => true,
+            Type::Numeric(_) => false,
+            // TODO: Look at if we need special handling for references
+            _ => {
+                unreachable!("ICE: expected array, slice, or numeric type but got {self:?}");
+            }
```

Comment:
Can you remove the catch-all here? I'd like this to error if we ever add/change types in the future.
It also seems odd to me to panic instead of handling the additional cases when the method is named so generally. Can you extend this to cover the missing cases as well instead of panicking?

Reply:
I removed the catch-all. I was originally unsure if I should handle references differently but I believe it is correct to just return false in this instance as this method is used for simply determining whether we should perform a read differently when we are returning a slice from an array_get or storing a slice in an array_set. A reference to a slice shouldn't be treated as a slice type itself.